docs/src/testing.md

```diff
@@ -0,0 +1,73 @@
+# Testing
+
+All solvers use the tests in this repository as extra correctness tests for themselves.
+If we find a bug in one solver, instead of adding a test to that particular repository, we add it here so that all solvers can benefit.
+
+## Example of adding a test
+
+For this test, we care about correctness, rather than performance.
+We don't expect solvers to efficiently decide that they have already solved the problem, only that calling optimize! twice doesn't throw an error or give the wrong answer.
+
+To give an example, ECOS errored calling `optimize!(model); optimize!(model)` ([ECOS.jl PR #72](https://github.com/jump-dev/ECOS.jl/pull/72)).
+We could add a test to ECOS.jl, but that would only stop us from re-introducing the bug to ECOS.jl in the future.
+Instead if we add a test here, then all solvers (e.g., SCS.jl, Gurobi.jl, Mosek.jl, ...) will also check that they handle a double optimize call!
+
+To resolve this issue, there are two steps:
+
+1. Add a new test to [src/Test/UnitTests/solve.jl](https://github.com/jump-dev/MathOptInterface.jl/blob/master/src/Test/UnitTests/solve.jl). The test should be something like
```

Comment:
Since the double-optimize error involves solving an optimization problem, add a new test ...

Reply:
done in PR #1189